source/base/exceptions.cc

```diff
@@ -522,6 +522,48 @@ namespace deal_II_exceptions
         }
         }
     }
+
+
+
+    std::string get_cusolver_error_string(cusolverStatus_t error_code)
+    {
+      std::string message;
+      switch (error_code)
+        {
+        case CUSOLVER_STATUS_NOT_INITIALIZED:
+        {
+          return "The cuSolver library was not initialized";
+        }
+        case CUSOLVER_STATUS_ALLOC_FAILED:
+        {
+          return "Resource allocation failed inside the cuSolver library";
+        }
+        case CUSOLVER_STATUS_INVALID_VALUE:
+        {
+          return "An unsupported value of parameter was passed to the function";
+        }
+        case CUSOLVER_STATUS_ARCH_MISMATCH:
+        {
+          return "The function requires a feature absent from the device architecture";
```

Comment:
Can we easily query the SM architecture here to add what the compiled version is? I know that one sets these in the makefile, but I have not checked if it is possible to query them at compile time. Or is this error message a hardware error not related to the compilation at all, so giving this info would not be useful at all? If I remember correctly, I have chosen different SM modes on Pascal at compile time.

Reply:
I am not sure what you mean here. The compilation in cuda is really in two phases: in the first phase, you decide which capability you want and in the second phase, you compile for the architecture. This allows you to compile with Kepler capabilities but run on Volta. I am not sure what would happen if you decide to compile for Fermi and run Volta but this error should never be triggered in practice. According to the documentation, this only happens for compute capability below 2.0 and we require a compute capability of 3.5. I've just added the error for completeness.

Reply:
Thanks for the clarification, your answer is precisely telling me what I wanted to know. :+1: